*1. msigdbr:*

  Dolgalev I (2025). _msigdbr: MSigDB Gene Sets for Multiple Organisms in a Tidy
  Data Format_. R package version 24.1.0,
  <https://CRAN.R-project.org/package=msigdbr>.

A BibTeX entry for LaTeX users is

  @Manual{,
    title = {msigdbr: MSigDB Gene Sets for Multiple Organisms in a Tidy Data Format},
    author = {Igor Dolgalev},
    year = {2025},
    note = {R package version 24.1.0},
    url = {https://CRAN.R-project.org/package=msigdbr},
  }

*2. edgeR:*

  Chen Y, Chen L, Lun ATL, Baldoni PL, Smyth GK (2025). edgeR v4: powerful
  differential analysis of sequencing data with expanded functionality and improved
  support for small counts and larger datasets. Nucleic Acids Research 53(2),
  gkaf018.

A BibTeX entry for LaTeX users is

  @Article{,
    author = {Yunshun Chen and Lizhong Chen and Aaron T L Lun and Pedro Baldoni and Gordon K Smyth},
    title = {{edgeR} v4: powerful differential analysis of sequencing data with expanded functionality and improved support for small counts and larger datasets},
    year = {2025},
    journal = {Nucleic Acids Research},
    volume = {53},
    number = {2},
    pages = {gkaf018},
    doi = {10.1093/nar/gkaf018},
  }

#*3. limma:*

  Ritchie, M.E., Phipson, B., Wu, D., Hu, Y., Law, C.W., Shi, W., and Smyth, G.K.
  (2015). limma powers differential expression analyses for RNA-sequencing and
  microarray studies. Nucleic Acids Research 43(7), e47.

A BibTeX entry for LaTeX users is

  @Article{,
    author = {Matthew E Ritchie and Belinda Phipson and Di Wu and Yifang Hu and Charity W Law and Wei Shi and Gordon K Smyth},
    title = {{limma} powers differential expression analyses for {RNA}-sequencing and microarray studies},
    journal = {Nucleic Acids Research},
    year = {2015},
    volume = {43},
    number = {7},
    pages = {e47},
    doi = {10.1093/nar/gkv007},
  }
> 

*4. TxDb.Hsapiens.UCSC.hg38.knownGene:* 

  Team BC, Maintainer BP (2024). _TxDb.Hsapiens.UCSC.hg38.knownGene: Annotation
  package for TxDb object(s)_. R package version 3.20.0.

A BibTeX entry for LaTeX users is

  @Manual{,
    title = {TxDb.Hsapiens.UCSC.hg38.knownGene: Annotation package for TxDb object(s)},
    author = {Bioconductor Core Team and Bioconductor Package Maintainer},
    year = {2024},
    note = {R package version 3.20.0},
  }

*5. GO.db:*

  Carlson M (2024). _GO.db: A set of annotation maps describing the entire Gene
  Ontology_. R package version 3.20.0.

A BibTeX entry for LaTeX users is

  @Manual{,
    title = {GO.db: A set of annotation maps describing the entire Gene Ontology},
    author = {Marc Carlson},
    year = {2024},
    note = {R package version 3.20.0},
  }

*6. Goseq:*

Young, M.D., Wakefield, M.J., Smyth, G.K., Oshlack, A., Gene ontology analysis for
  RNA-seq: accounting for selection bias, Genome Biology, 11, 2, Feb 2010, R14

A BibTeX entry for LaTeX users is

  @Article{,
    title = {Gene ontology analysis for RNA-seq: accounting for selection bias},
    author = {Matthew D Young and Matthew J Wakefield and Gordon K Smyth and Alicia Oshlack},
    journal = {Genome Biology},
    volume = {11},
    pages = {R14},
    year = {2010},
  }

*7. Rsubread:*

  Liao Y, Smyth GK and Shi W (2019). The R package Rsubread is easier, faster,
  cheaper and better for alignment and quantification of RNA sequencing reads.
  Nucleic Acids Research 47(8), e47.

A BibTeX entry for LaTeX users is

  @Article{,
    title = {The {R} package {Rsubread} is easier, faster, cheaper and better for alignment and quantification of {RNA} sequencing reads},
    author = {Yang Liao and Gordon K. Smyth and Wei Shi},
    journal = {Nucleic Acids Research},
    year = {2019},
    volume = {47},
    issue = {8},
    pages = {e47},
    doi = {10.1093/nar/gkz114},
  }

  *8. Rsamtools:*

  Morgan M, Pagès H, Obenchain V, Hayden N (2024). _Rsamtools: Binary alignment
  (BAM), FASTA, variant call (BCF), and tabix file import_.
  doi:10.18129/B9.bioc.Rsamtools <https://doi.org/10.18129/B9.bioc.Rsamtools>, R
  package version 2.22.0, <https://bioconductor.org/packages/Rsamtools>.

A BibTeX entry for LaTeX users is

  @Manual{,
    title = {Rsamtools: Binary alignment (BAM), FASTA, variant call (BCF), and tabix
file import},
    author = {Martin Morgan and Hervé Pagès and Valerie Obenchain and Nathaniel Hayden},
    year = {2024},
    note = {R package version 2.22.0},
    url = {https://bioconductor.org/packages/Rsamtools},
    doi = {10.18129/B9.bioc.Rsamtools},
  }
  
*9. DESeq2:*

  Love, M.I., Huber, W., Anders, S. Moderated estimation of fold change and
  dispersion for RNA-seq data with DESeq2 Genome Biology 15(12):550 (2014)

A BibTeX entry for LaTeX users is

  @Article{,
    title = {Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2},
    author = {Michael I. Love and Wolfgang Huber and Simon Anders},
    year = {2014},
    journal = {Genome Biology},
    doi = {10.1186/s13059-014-0550-8},
    volume = {15},
    issue = {12},
    pages = {550},
  }

*10. KEGGREST:*

  Tenenbaum D, Maintainer B (2024). _KEGGREST: Client-side REST access to the Kyoto
  Encyclopedia of Genes and Genomes (KEGG)_. doi:10.18129/B9.bioc.KEGGREST
  <https://doi.org/10.18129/B9.bioc.KEGGREST>, R package version 1.46.0,
  <https://bioconductor.org/packages/KEGGREST>.

A BibTeX entry for LaTeX users is

  @Manual{,
    title = {KEGGREST: Client-side REST access to the Kyoto Encyclopedia of Genes and
Genomes (KEGG)},
    author = {Dan Tenenbaum and Bioconductor Package Maintainer},
    year = {2024},
    note = {R package version 1.46.0},
    url = {https://bioconductor.org/packages/KEGGREST},
    doi = {10.18129/B9.bioc.KEGGREST},
  }

*11. EnhancedVolcano:*

  Blighe K, Rana S, Lewis M (2024). _EnhancedVolcano: Publication-ready volcano
  plots with enhanced colouring and labeling_. doi:10.18129/B9.bioc.EnhancedVolcano
  <https://doi.org/10.18129/B9.bioc.EnhancedVolcano>, R package version 1.24.0,
  <https://bioconductor.org/packages/EnhancedVolcano>.

A BibTeX entry for LaTeX users is

  @Manual{,
    title = {EnhancedVolcano: Publication-ready volcano plots with enhanced colouring and
labeling},
    author = {Kevin Blighe and Sharmila Rana and Myles Lewis},
    year = {2024},
    note = {R package version 1.24.0},
    url = {https://bioconductor.org/packages/EnhancedVolcano},
    doi = {10.18129/B9.bioc.EnhancedVolcano},
  }
  
*12. pathview:* 

  Luo, W. and Brouwer C., Pathview: an R/Bioconductor package for pathway-based data
  integration and visualization. Bioinformatics, 2013, 29(14): 1830-1831, doi:
  10.1093/bioinformatics/btt285

A BibTeX entry for LaTeX users is

  @Article{,
    author = {{Luo} and {Weijun} and {Brouwer} and {Cory}},
    title = {Pathview: an R/Bioconductor package for pathway-based data integration and visualization},
    journal = {Bioinformatics},
    year = {2013},
    doi = {10.1093/bioinformatics/btt285},
    volume = {29},
    number = {14},
    pages = {1830-1831},
  }
